modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/expression/MBString.java

```diff
@@ -17,32 +17,38 @@
 
 package org.geotools.mbstyle.expression;
 
-import org.geotools.factory.CommonFactoryFinder;
 import org.geotools.mbstyle.parse.MBFormatException;
 import org.geotools.mbstyle.parse.MBObjectParser;
 import org.json.simple.JSONArray;
-import org.opengis.filter.FilterFactory2;
 import org.opengis.filter.expression.Expression;
+import org.opengis.filter.expression.Function;
 
-public class MBString extends MBExpression{
+import java.util.ArrayList;
+import java.util.List;
 
+public class MBString extends MBExpression{
+    MBObjectParser parse;
     public MBString(JSONArray json){
         super(json);
+        parse = new MBObjectParser(MBString.class);
     }
 
-
     /**
      * Returns a string consisting of the concatenation of the inputs.
      * Example: ["concat", string, string, ...]: string
      * @return
      */
     public Expression stringConcat(){
-        String concatString = "";
+        List<Expression> list = new ArrayList<>();
         for(int i =1; i < json.size(); i++) {
-            String s = (String) json.get(i);
-            concatString += s;
+            Object o = parse.get(json, i);
```

Comment:
I imagine you'll be repeating this structure a lot when implementing Expressions and wonder if there is any way of simplifying some part of it. Perhaps a utility function in MBExpression that parses the JSON array and returns a list of the Expressions representing each value? (You'll still need some way of verifying the expressions are of the right type - in this case, strings)

Reply:
This one actually might be somewhat tough to simplify looking into it.  The MBObjectParser has different functions for different data types, ie. parse.value(...) for numbers, parse.bool(...) for boolean, parse.get(...) for strings etc. and most of the MBExpressions have a set amount of arguments that they take, with a few allowing variable args.

It would be easy enough to parse the arguments out if there weren't a need for the arguments themselves to be evaluated by the MBObjectParser for nested expressions.

Any additional thoughts would be appreciated.